src/main.rs

```diff
@@ -5,16 +5,26 @@ mod setup_scripts;
 pub mod tests;
 pub mod utils;
 
-use clap::{ArgAction, Parser};
+use std::fs::File;
```

Comment:
More of a general comment but I think the code would gain clarity if the functions/structs only accepts clearly the arguments they need, instead of passing `args` and `config` everywhere, it makes it harder to understand what composes each structs/what functions uses without diving in the implementation

Reply:
I will add this in the refactor PR that we are planning